src/raft.rs

```diff
@@ -1993,16 +1983,23 @@ impl<T: Storage> Raft<T> {
     fn check_quorum_active(&mut self) -> bool {
         let self_id = self.id;
         let mut act = 0;
-        for (&id, pr) in self.mut_prs().iter_mut() {
-            if id == self_id {
-                act += 1;
-                continue;
+
+        self.prs = if let Some(mut prs) = self.prs.take() {
+            for (&id, pr) in prs.voters_mut() {
+                if id == self_id || pr.recent_active {
```

Comment:
`recent_active` should be set to false before continue.

Reply:
Then we are back to the same code as before https://github.com/pingcap/raft-rs/pull/119/files/2fa4a6543e95d75ba929dc12bb3c8eaa274eec98..47e441f9f7c5db5e021fd13a831b99587c726b9b#r222180743

```rust
            for (&id, pr) in prs.voters_mut() {
                if id == self_id || pr.recent_active {
                    act += 1;
                    pr.recent_active = false;
                    continue;
                }
                pr.recent_active = false;
            }
```

Which is equivalent to:

```rust
            for (&id, pr) in prs.voters_mut() {
                if id == self_id || pr.recent_active {
                    act += 1;
                }
                pr.recent_active = false;
            }
```

Reply:
Apparently not. You can compare the code with original especially how `recent_active` is mutated when it's self_id or not.

Reply:
Hm yes, I took a very close look and you're right. I have fixed this.